packages/ds-react-core/src/ui/Button/Button.tsx

```diff
@@ -1,39 +1,54 @@
 import type React from "react";
 
-import "./button.css";
+import "./styles.css";
+
+// TODO: this is how appearance could work as enum
+//
+// export enum ButtonAppearance {
+//   DEFAULT = "default",
+//   BASE = "base",
+//   POSITIVE = "positive",
+//   NEGATIVE = "negative",
+//   LINK = "link",
+// }
 
 export interface ButtonProps {
-  /** Is this the principal call to action on the page? */
-  primary?: boolean;
-  /** What background color to use */
-  backgroundColor?: string;
-  /** How large should the button be? */
-  size?: "small" | "medium" | "large";
+  /* A unique identifier for the button */
+  id?: string;
+  /** Additional CSS classes */
+  className?: string;
+  /** The visual style of the button */
+  appearance?: "default" | "base" | "positive" | "negative" | "link";
   /** Button contents */
   label: string;
   /** Optional click handler */
   onClick?: () => void;
 }
 
-/** Primary UI component for user interaction */
+// combine custom props with all native button element attributes
+export type ButtonPropsType = ButtonProps &
```

Comment:
Whenever we spread the props onto HTML element we will have to do this, otherwise TS will complain.

I guess we will have to see if spreading will be part of all the components, or just the ones that build on top of native ones (button, inputs, etc).
There is probably a bit less of a reason to spread props on "divs" around components like accordion, modal, or what not.